doc/en/components/grids/_shared/column-pinning.md

```diff
@@ -0,0 +1,503 @@
+
+```html
+<ng-template igxHeader let-column #pinTemplate>
+    <div class="title-inner">
+        <span style="float:left">{{column.header || column.field}}</span>
+        <igx-icon class="pin-icon" [class.pinned]="column.pinned" [class.unpinned]="!column.pinned" fontSet="fas" name="fa-thumbtack"
+            (click)="toggleColumn(column)"></igx-icon>
+    </div>
+</ng-template>
+<igx-hierarchical-grid class="hierarchicalGrid" [data]="localdata" [autoGenerate]="false"
+    [height]="'500px'" [width]="'100%'" #hierarchicalGrid>
+    <igx-column field="CompanyName" header="Company Name" [headerTemplate]="pinTemplate" width="200px" [pinned]="true"></igx-column>
+    <igx-column field="ContactName" header="Contact Name" [headerTemplate]="pinTemplate" width="150px"></igx-column>
+    <igx-column field="ContactTitle" header="Contact Title" [headerTemplate]="pinTemplate" width="200px"></igx-column>
+    <igx-row-island [key]="'Orders'" [autoGenerate]="false">
+        <igx-column field="OrderDate" header="Order Date" [headerTemplate]="pinTemplate" [dataType]="'date'" width="150px"></igx-column>
+        <igx-column field="RequiredDate" header="Required Date" [headerTemplate]="pinTemplate" [dataType]="'date'" width="150px"></igx-column>
+        <igx-column field="ShippedDate" header="Shipped Date" [headerTemplate]="pinTemplate" [dataType]="'date'" width="150px"></igx-column>
+        <igx-column field="ShipVia" header="Ship Via" [headerTemplate]="pinTemplate" width="150px"></igx-column>
+        <igx-row-island [key]="'OrderDetails'" [autoGenerate]="false">
+            <igx-column field="UnitPrice" header="Unit Price" width="150px"></igx-column>
+            <igx-column field="Quantity" width="150px"></igx-column>
+            <igx-column field="Discount" width="150px"></igx-column>
+        </igx-row-island>
+    </igx-row-island>
+</igx-hierarchical-grid>
+```
+
+<!-- ComponentEnd: HierarchicalGrid -->
+
+On click of the custom icon the pin state of the related column can be changed using the column's API methods.
+
+```typescript
+public toggleColumn(col: IgxColumnComponent) {
+    col.pinned ? col.unpin() : col.pin();
+}
+```
+
+<!-- end: Angular -->
+
+### Demo
+
+<!-- ComponentStart: Grid -->
+
+<code-view style="height:510px" 
+           data-demos-base-url="{environment:demosBaseUrl}" 
+           iframe-src="{environment:demosBaseUrl}/{ComponentSample}-sample-pinning" >
+</code-view>
+
+<!-- ComponentEnd: Grid -->
+<!-- ComponentStart: TreeGrid -->
+
+<code-view style="height:630px" 
+           data-demos-base-url="{environment:demosBaseUrl}" 
+           iframe-src="{environment:demosBaseUrl}/{ComponentSample}-column-pinning" >
+</code-view>
+
+<!-- ComponentEnd: TreeGrid -->
+<!-- ComponentStart: HierarchicalGrid -->
+
+<code-view style="height:510px" 
+           data-demos-base-url="{environment:demosBaseUrl}" 
+           iframe-src="{environment:demosBaseUrl}{ComponentSample}-pinning" >
+</code-view>
```

Comment:
remove 2 out of 3 code-view blocks and Component build flags because will use the same routing path for Grid, TreeGrid, HierarchicalGrid in samples.